index.ts

```diff
@@ -0,0 +1,60 @@
+const products = require("./data/products.json");
+
+interface Product {
+  name: string;
+  description: string;
+  price: number;
+  image: URL;
+  limit: number;
+  amount: number;
+}
+
+interface Cart {
+  products: Product[];
+}
+
+const addProduct = (cart: Cart, product: Product) => {
+  cart.products.push(product);
+};
+
+const removeProduct = (cart: Cart, productToRemove: Product) => {
+  cart.products.forEach((product, i) => {
+    if (product.name === productToRemove.name) cart.products.splice(i, 1);
+  });
+};
+
+const updateProductAmount = (
+  cart: Cart,
+  productToChangeAmount: Product,
+  amount: number
+) => {
+  cart.products.forEach((product, i) => {
```

Comment:
read about array functions. use relevant function

Reply:
not sure what array function you meant, did it now with map and it looks like a good solution but as good as the previous.